src/audio/module_adapter/module/iadk_modules.c

```diff
@@ -63,8 +63,8 @@ static int iadk_modules_init(struct processing_module *mod)
 
 	mod_cfg.data = md->cfg.data;
 	/* Intel modules expects DW size here */
-	mod_cfg.size = (md->cfg.size >> 2);
-	md->private = md->cfg.data;
+	mod_cfg.size = md->cfg.size >> 2;
+	md->private = mod;
```

Comment:
@lyakh is this really correct? mod is the dev's private data already. What the IPC config data bring is really the module's config data. So the original code seems more intuitive to me. @jxstelter what do you think?

Reply:
I don't understand this change. The problem here is that for built in modules you can store IPC configuration inside module data. IADK modules don't do this and require to have configuration stored somewhere. I need to step back to find out details, but storing IPC4 configuration data was here required. 

Additionally, please look at implementation of module_adapter_get_attribute() for IPC4:
https://github.com/lyakh/sof/blob/185f80d7735a3022429d4f6d833db4246cec48de/src/audio/module_adapter/module_adapter.c#L981
I think it will not work correct anymore after this change.

Reply:
> Additionally, please look at implementation of module_adapter_get_attribute() for IPC4:
> https://github.com/lyakh/sof/blob/185f80d7735a3022429d4f6d833db4246cec48de/src/audio/module_adapter/module_adapter.c#L981
> I think it will not work correct anymore after this change.

@jxstelter ah, that's where it's used - I didn't find it. Then yes, this would break that. But as is it's broken as well, because that `.data` pointer at least in some cases points to the host IPC memory window, where content is overwritten by the host. So we do indeed need to allocate a buffer for that configuration

Reply:
@ranj063 @jxstelter looking at that `module_adapter_get_attribute()` more - why is it returning the `.private` pointer? That pointer is used by various modules for their internal data, e.g. by eq_fiir: https://github.com/thesofproject/sof/blob/8083eb0156a56ac43d31424d0183dd1993218c19/src/audio/eq_fir/eq_fir.c#L351 and it's returned by `module_get_private_data()` - isn't that `module_adapter_get_attribute()` actually wrong?